### PR TITLE
Adding tests to nvme-cli test

### PR DESCRIPTION
**Pull request by narasimhan-v**
Added the following tests to nvme-cli test.
*) compare
*) flush
*) writezeroes
*) writeuncorrectable
*) dsm
*) reset
*) subsystemreset

Made sure installation of nvme-cli is skipped,
if latest nvme is installed already.

Fetching nvme-cli release tarball instead of git.

Signed-off-by: Narasimhan V <sim@linux.vnet.ibm.com>

### CODE REVIEW

**Comment by narasimhan-v** [comment]
[job_log.txt](https://github.com/avocado-framework-tests/avocado-misc-tests/files/524460/job_log.txt)

File: io/disk/ssd/nvmetest.py [hunk 1]
```diff
@@ -44,11 +44,19 @@ def setUp(self):
...
-        nvme_link = 'https://github.com/linux-nvme/nvme-cli.git'
-        git.get_repo(nvme_link, destination_dir=nvme_path)
-        build.make(nvme_path)
-        build.make(nvme_path, extra_args='install')
+        locations = ["https://github.com/linux-nvme/nvme-cli/archive/"
+                     "master.zip"]
+        tarball = self.fetch_asset("nvme-cli.zip", locations=locations,
+                                   expire='15d')
+        archive.extract(tarball, self.srcdir)
+        process.system("ls %s" % self.srcdir)
```
Comment (PraveenPenguin) [inline io/disk/ssd/nvmetest.py]: is this is left over ?

File: io/disk/ssd/nvmetest.py [hunk 2]
```diff
@@ -44,11 +44,19 @@ def setUp(self):
...
+        os.chdir("%s/nvme-cli-master" % self.srcdir)
+        process.system("./NVME-VERSION-GEN", ignore_status=True)
+        if process.system_output("cat NVME-VERSION-FILE").strip("\n").\
+            split()[-1] != process.system_output("nvme version").\
+                strip("\n").split()[-1]:
```
Comment (PraveenPenguin) [inline io/disk/ssd/nvmetest.py]: ok , cool

File: io/disk/ssd/nvmetest.py (hunk 1 above)
Comment (narasimhan-v) [inline io/disk/ssd/nvmetest.py]: Oops, yes. Please :-)

**Pull request merged**